tools/analysis/runtime.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited, Google, and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import trappy
+import numpy as np
+import argparse
+
+class RunData:
+    def __init__(self, pid, comm, time):
+        self.pid = pid
+        self.comm = comm
+        self.last_start_time = time
+        self.total_time = np.float64(0.0)
+        self.start_time = -1
+        self.end_time = -1
+        self.running = 1
+        self.maxrun = -1
+
+parser = argparse.ArgumentParser(description='Analyze runtimes')
+
+parser.add_argument('--trace', dest='trace_file', action='store', required=True,
```

Comment:
action='store' is not needed since this is the default (and is expected)

Reply:
Are "dest=" arguments needed since they are bound to a usual variable before being used by the script ? Renaming already happens there. For the ones that are not renamed, a default dest is computed using the long option name, replacing dashes by underscores